### PR TITLE
Feature/backports.csv

### PR DESCRIPTION
**Pull request by ickc**
@reenberg, I'm "stealing" this to check how well it works. Thanks.

### CODE REVIEW

**Comment by reenberg** [comment]
@ickc Any reason why you mention v20.7.0?  
As noted in the code, v20.2.2 is the minimum required version for the environment markers to work, according to my test.  And then I just chose v20.6.8 as the version, since it seemed to have a few fixes to the environment markers in the version up until that.  Though it seemed to work just fine with v20.2.2.

Do you have any objections against depending on such a "new" version of setuptools?

The reason that 15-19 is needed is because running `python setup.py install` will fail if your version of setuptools is < v20.2.2 as it can't parse the syntax.  And thus you can't rely on it to install the required version of setuptools.  I tried briefly to look into using the setup_requires keyword, however it seems mainly to be used for distutils extensions.  But it should probably be investigated further.

**Comment by reenberg** [comment]
I made an attempt at fixing the tests.  Now they all pass for py2 and py3.  

I basically changed all the string to be unicode in py2 by adding `u` in front of all the strings, which should have no effect in py3(?)

I also fixed a division issue that made some tests fail in py2 due to the difference with py2 and py3 division.  

And i re-added the `str()` around the `include` parameter in `read_data` so that the test that tries to read from `True` doesn't fail.


However travis-ci still fail for all py3.  `make test` didn't report any error here when running in a py3 virtual environment.

**Comment by reenberg** [comment]
Ah `make test` doesn't run pep8.. Did not notice you had to run `make testFull` in order to get that.

Opps, two debug lines made it into the commit :(

File: pantable/pantable.py [hunk 1]
```diff
@@ -177,14 +182,13 @@ def read_data(include, data):
     Return None when the include path is invalid.
     """
     if include is None:
-        if py2:
-            data = data.encode('utf-8')
```
Comment (ickc) [inline pantable/pantable.py]: Would this line means that you don't need to change the tests to start with `u'...'`?

File: tests/pantable/test_auto_width.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 """
 """
+from __future__ import division
```
Comment (ickc) [inline tests/pantable/test_auto_width.py]: This probably is not needed. The reason is that I use `pasteurize` to the py2 wheel which add this automatically.

File: pantable/pantable.py [hunk 3]
```diff
@@ -41,14 +41,19 @@
 ```
 """
 
-import csv
+from __future__ import division
```
Comment (ickc) [inline pantable/pantable.py]: Same here.

File: pantable/pantable.py [hunk 4]
```diff
@@ -177,14 +182,13 @@ def read_data(include, data):
...
-        io_universal = io.BytesIO if py2 else io.StringIO
-        with io_universal(data) as file:
+        with io.StringIO(data) as file:
             raw_table_list = list(csv.reader(file))
     else:
         try:
-            with open(str(include)) as file:
+            # Handle bad yaml meta-data input by trying to make a string out of
+            # include, so we can open it as a file or get an IOError.
+            with io.open(str(include)) as file:
```
Comment (ickc) [inline pantable/pantable.py]: On python3, open is the same as io.open.

You might need this on Python2, but `pasteurize` probably done this automatically.

**Comment by ickc** [comment]
I added some notes on pasteurize in #23, which will be merged soon and show up in [pantable/CONTRIBUTING.md at master · ickc/pantable](https://github.com/ickc/pantable/blob/master/CONTRIBUTING.md).

**Comment by ickc** [comment]
Merging to another branch and I'll refactor this.

**Pull request merged**